CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(funchook C ASM)
+
+include(CheckSymbolExists)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
+
+if (NOT DEFINED _GNU_SOURCE)
+    check_symbol_exists(__GNU_LIBRARY__ "features.h" _GNU_SOURCE)
+
+    if (NOT _GNU_SOURCE)
+        unset(_GNU_SOURCE CACHE)
+        check_symbol_exists(_GNU_SOURCE "features.h" _GNU_SOURCE)
+    endif ()
+endif ()
```

Comment:
Could you remove code related to `_GNU_SOURCE` in this file and add the following code just before `#include <stdio.h>` in funchook_unix.c as in src/funchook_io.c and src/os_func_unix.c？
```
#if defined __linux && !defined(_GNU_SOURCE)
#define _GNU_SOURCE
#endif
```